File: controller/json/delete_item.php
<?php
namespace Core;
use Exception;

if (!defined('__GOOSE__')) exit();

/**
 * delete json
 *
 * @var Goose $this
 */

try
{
  // check and set srl
  $srl = (int)$this->params['srl'];
  if (!($srl && $srl > 0))
  {
    throw new Exception(Message::make('error.notFound', 'srl'));
  }

  // connect db
  $this->model->connect();

  // check access
  $token = Controller::checkAccessItem((object)[
    'model' => $this->model,
    'table' => 'json',
    'srl' => $srl,
  ]);

  // remove item
  $output = Controller::delete((object)[
    'model' => $this->model,
    'table' => 'json',
    'srl' => $srl,
  ]);

  // set output
  if ($token) $output->_token = $token->jwt;

  // disconnect db
  $this->model->disconnect();

  // output data
  Output::data($output);
}
catch (Exception $e)
{
  $this->model->disconnect();
  Error::data($e->getMessage(), $e->getCode());
}

File: src/endpoints/auth/token/delete_item.py
from . import __types__ as types
from src import output
from src.libs.db import DB, Table
from src.modules.verify import checking_token

async def delete_item(params: dict = {}, req = None, _db: DB = None, _check_token = True):

    # set values
    result = None
    db = _db if _db else DB().connect()

    try:
        # set params
        params = types.DeleteItem(**params)

        # checking token
        if _check_token: checking_token(req, db)

        # get item
        item = db.get_item(
            table_name=Table.TOKEN.value,
            where=[
                f'srl = {params.srl}',
                f'AND expires IS NULL'
            ],
        )
        if not item: raise Exception('Item not found', 400)

        # delete data
        db.update_item(
            table_name=Table.TOKEN.value,
            where=[ f'srl = {params.srl}' ],
            placeholders=[ 'expires = :expires' ],
            values={ 'expires': 0 },
        )

        # set result
        result = output.success({
            'message': 'The token has expired.',
        }, _req=req)
    except Exception as e:
        result = output.exc(e, _req=req)
    finally:
        if not _db and db: db.disconnect()
        return result
